dev_guide/application_health.adoc

```diff
@@ -11,31 +11,125 @@ toc::[]
 
-// TODO: example of liveness check
+[cols="2", width="80%"]
+|===
 
-== High-level application health checks
+|Liveness Probe
+|A liveness probe checks if the container in which it is configured is still
+running. If the liveness probe fails, the kubelet kills the container, which
+will be subjected to its restart policy. Set a liveness check by configuring the
+`*template.spec.containers.livenessprobe*` stanza of a pod configuration.
 
-Holistic notions of application health that cut across multiple components cannot be handled with
-container scoped liveness checks.  It is possible for each component in a distributed application
-to be healthy and responsive but for the overall system behavior to be unhealthy.  Administrators
-of distributed systems use health checks to test the behavior of a overall health of a system.  A
-typical example of this type of check is a script which performs a test transaction: something
-which exercises the components of the system together and verifies that the high-level behavior of
-the system in the context of the test transaction matches the expected behavior of a health system.
+|Readiness Probe
+|A readiness probe determines if a container is ready to service requests. If
+the readiness probe fails a container, the endpoints controller ensures the
+container has its IP address removed from the endpoints of all services. A
+readiness probe can be used to signal to the endpoints controller that even
+though a container is running, it should not receive any traffic from a proxy.
+Set a readiness check by configuring the
+`*template.spec.containers.readinessprobe*` stanza of a pod configuration.
+
+|===
+
+Both probes can be configured in three ways:
+
+*HTTP Checks*
+
+The kubelet uses a web hook to determine the healthiness of the container. The
+check is deemed successful if the hook returns with 200 or 399. The following is
+an example of a readiness check using the HTTP checks method:
+
+.Readiness HTTP check
+====
+----
+...
+readinessProbe:
+  httpGet:
+    path: /healthz
+    port: 8080
+  initialDelaySeconds: 15
+  timeoutSeconds: 1
+...
+----
+====
+
+A HTTP check is ideal for complex applications that can return with a 200 status
+when completely initialized.
+
+*Container Execution Checks*
+
+The kubelet executes a command inside the container. Exiting the check with
+status 0 is considered a success. The following is an example of a liveness
+check using the container execution method:
+
+.Liveness Container Execution Check
+====
+----
+...
+livenessProbe:
+  exec:
+    command:
+    - cat
+    - /tmp/health
+  initialDelaySeconds: 15
+  timeoutSeconds: 1
+...
+----
+====
+
+*TCP Socket Checks*
+
+The kubelet attempts to open a socket to the container. The container is only
+considered healthy if the check can establish a connection. The following is an
+example of a liveness check using the TCP socket check method:
+
+.Liveness TCP Socket Check
+====
+----
+...
+livenessProbe:
+  tcpsocket
```

Comment:
Yeah those are the docs I used to write it up. Done. Also, camel case is a weird term. :camel: